job_creator.php

```diff
@@ -42,6 +42,23 @@ public function getInstallerVersion(): string
                 return $cmsMajor . '.' . $portions[1] . '.x-dev';
             }
         }
+        // has a lockstepped .x-dev requirement in composer.json
+        if (file_exists($this->composerJsonPath)) {
+            $json = json_decode(file_get_contents($this->composerJsonPath));
+            foreach (LOCKSTEPPED_REPOS as $lockedSteppedRepo) {
+                $composerRepo = 'silverstripe/' . str_replace('silverstripe-', '', $lockedSteppedRepo);
+                if (isset($json->require->{$composerRepo})) {
+                    $version = $json->require->{$composerRepo};
+                    if (preg_match('#^([0-9\.]+)\.x\-dev$#', $version, $matches)) {
+                        $versionNumber = $matches[1];
+                        if (in_array($lockedSteppedRepo, LOCKSTEPPED_REPOS_VERSION_IS_THREE_LESS)) {
```

Comment:
```suggestion
                        // If the lockstepped dependency is "three less" (e.g. silverstripe/admin is 3 major versions behind silverstripe/installer), account for that here.
                        if (in_array($lockedSteppedRepo, LOCKSTEPPED_REPOS_VERSION_IS_THREE_LESS)) {
```
Just explaining a little bit more about this "three less" thing, since it's not really intuitive.

Reply:
Updated